File: testdata/precision-suite-php/tp_xss.php
<?php
// Reflected XSS: a request parameter is echoed straight into the HTML response
// with no output encoding, so an attacker's markup/script executes in the
// victim's browser. CWE-79, TAINT-003.

function greet()
{
    $name = $_GET['name'];
    echo "<h1>Hello, " . $name . "</h1>"; // nox-expect: TAINT-003
}

File: testdata/precision-suite-scala/tp_xss.scala
// Reflected XSS: a request parameter is wrapped in play.twirl.api.Html, which
// marks the string as trusted markup and bypasses Twirl's auto-escaping — CWE-79.
// A correct scanner fires TAINT-003. Because the value is reflected into the HTML
// response without encoding, an attacker-supplied `<script>` executes.
package controllers

import play.twirl.api.Html

class GreetingController {
  // greet reflects an un-encoded request value into trusted HTML — the bug.
  def greet(request: Request): Html = {
    val name = request.getQueryString("name")
    Html("<h1>Hello " + name + "</h1>") // nox-expect: TAINT-003
  }
}
